src/utils/jwt.utils.ts

```diff
@@ -22,3 +26,31 @@ export class jwtUtility {
     });
   }
 }
+
+//! The below utility is only used for resetting the password
+export class jwtUtility2 {
+  static async generateToken(userData: string): Promise<string> {
+    const userSecret = crypto.randomUUID();
+    await User.update(
+      {
+        jwtSecret: userSecret,
+      },
+      {
+        where: {
+          email: userData,
+        },
+      },
+    );
+    return JWT.sign({ userData }, userSecret, { expiresIn: 300 });
+  }
+
+  static verifyToken(token: string, userSecret: string): any | Error {
+    return JWT.verify(token, userSecret, (err, decoded) => {
+      if (err) {
+        throw new Error(err.message);
+      }
+
+      return decoded;
+    });
+  }
+}
```

Comment:
```suggestion
```

refactor the whole file and have re-usable functions. check the example below and go along with it. 

```
import jwt from "jsonwebtoken";
import dotenv from "dotenv";
import crypto from "crypto";
import User from "../database/models/User.model";

dotenv.config();

const secretToken = process.env.SECRET_TOKEN;
if (!secretToken) {
  throw new Error("SECRET_TOKEN environment variable not set");
}

interface TokenOptions {
  expiresIn?: string;
}

function generateToken(payload: string, secret: string, options?: TokenOptions): string {
  return jwt.sign(payload, secret, options);
}

function verifyToken(token: string, secret: string): any | Error {
  // Have the Verify Token Logic Here
}

async function generatePasswordResetToken(userData: string, userSecret: string): Promise<string> {
 // Have generatePasswordResetToken Logic here
  const options: TokenOptions = { expiresIn: "5m" };
  return generateToken({ userData }, userSecret, options);
}

function verifyPasswordResetToken(token: string, userSecret: string): any | Error {
  // Have the verifyPasswordResetToken Logic here
}

export {
  generateToken,
  verifyToken,
  generatePasswordResetToken,
  verifyPasswordResetToken,
};
```